lib/Jsrt/JsrtDebuggerObject.cpp

```diff
@@ -484,26 +487,36 @@ Js::DynamicObject* JsrtDebuggerStackFrame::Evaluate(Js::ScriptContext* scriptCon
                 resolvedObject.obj = scriptContext->GetLibrary()->GetUndefined();
             }
         }
+        else
+        {
+          success = true;
+        }
+
         if (resolvedObject.obj != nullptr)
         {
             resolvedObject.scriptContext = scriptContext;
 
             charcount_t len = Js::JavascriptString::GetBufferLength(source);
             resolvedObject.name = AnewNoThrowArray(this->debuggerObjectsManager->GetDebugObjectArena(), WCHAR, len + 1);
-            if (resolvedObject.name == nullptr)
+
+            if (resolvedObject.name != nullptr)
+            {
+                wcscpy_s((WCHAR*)resolvedObject.name, len + 1, source);
+            }
+            else
             {
-                return nullptr;
+                // len can be big, if we failed just have empty string
+                resolvedObject.name = _u("");
             }
-            wcscpy_s((WCHAR*)resolvedObject.name, len + 1, source);
 
             resolvedObject.typeId = Js::JavascriptOperators::GetTypeId(resolvedObject.obj);
             JsrtDebuggerObjectBase::CreateDebuggerObject<JsrtDebuggerObjectProperty>(this->debuggerObjectsManager, resolvedObject, scriptContext, [&](Js::Var marshaledObj)
             {
-                evalResult = (Js::DynamicObject*)marshaledObj;
+                *evalResult = (Js::DynamicObject*)marshaledObj;
```

Comment:
No because here we can reach from both an script exception and successful execution.